MINOR: Remove a couple of redundant `CoreUtils.rm` methods

ijuma opened the pull request:
Also:
- Rename remaining `CoreUtils.rm` to `delete` for consistency
- Use `try with resources` in `Utils` to simplify code
- Silence compiler warning due to exception catch clause in `TestUtils`

ijuma committed:
Remove a couple of  `CoreUtils.rm` methods

Use `Utils.delete` instead.

ijuma committed:
Rename remaining `CoreUtils.rm` to `delete` for consistency

ijuma committed:
Use try with resources in `Utils`

ijuma committed:
Silence compiler warning by adding `: Throwable` to catch clause

ijuma commented:
Review by @guozhangwang please.

guozhangwang commented on core/src/test/scala/unit/kafka/log/BrokerCompressionTest.scala:
```diff
@@ -26,9 +25,11 @@ import org.junit.Assert._
 import org.junit.runners.Parameterized.Parameters
-import java.util.{Properties, Collection, ArrayList}
-import kafka.server.KafkaConfig
+import java.util.{Collection, Properties}
```
nit: move this line after the kafka clients package imports?

guozhangwang commented on core/src/test/scala/unit/kafka/server/HighwatermarkPersistenceTest.scala:
```diff
@@ -25,9 +26,10 @@ import org.junit._
 import java.util.concurrent.atomic.AtomicBoolean
-import org.apache.kafka.common.utils.{MockTime => JMockTime}
+
+import org.apache.kafka.common.utils.{Utils, MockTime => JMockTime}
```
nit: Move this import before java imports?

guozhangwang commented on core/src/test/scala/unit/kafka/server/OffsetCommitTest.scala:
```diff
@@ -19,20 +19,19 @@ package kafka.server
 import java.util.Properties
 import java.io.File
 
+import org.apache.kafka.common.utils.Utils
```
nit: Move this line after line 26?

guozhangwang commented:
LGTM except a few very minor comments.

ijuma committed:
Organise imports

As suggested by Guozhang.

ijuma commented:
@guozhangwang Addressed the review comments. I wasn't aware that we group java.\* imports after other imports. Interesting, in my experience I've seen the opposite (java.\* imports first).

guozhangwang commented:
@ijuma There are some discrepancies, yes. But the general idea is to group all kafka.\* imports, and all o.a.kafka.\* imports, and all common java / scala imports along with external dependencies like jackson / log4j.

guozhangwang commented:
Merged to trunk and 0.10.0.
